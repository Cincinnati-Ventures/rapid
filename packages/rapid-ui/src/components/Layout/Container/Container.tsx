import React from 'react';
import { RapidStyles } from '../../../utils';

type MaxWidth = 'sm' | 'md' | 'lg';

interface ContainerProps extends React.HTMLAttributes<HTMLDivElement> {
	styles?: string;
	maxWidth?: MaxWidth;
}

const parseMaxWidth = (maxWidth: MaxWidth) => {
	switch (maxWidth) {
		case 'sm':
			return 'max-w-5xl';
		case 'md':
			return 'max-w-7xl';
		case 'lg':
			return 'max-w-container';
	}
};

const Container = React.forwardRef<HTMLDivElement, ContainerProps>(
	({ styles, maxWidth = 'lg', ...rest }, ref) => {
		const defaultStyles = `container ${parseMaxWidth(
			maxWidth,
		)} mx-auto px-4`;
		return (
			<div
				ref={ref}
				{...rest}
				className={RapidStyles(styles || rest.className, defaultStyles)}
			/>
		);
	},
);

export default Container;
