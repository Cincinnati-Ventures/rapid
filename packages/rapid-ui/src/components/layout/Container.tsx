import React from 'react';
import { RapidStyles } from '../../utils';

type MaxWidth = 'sm' | 'md' | 'lg';

interface ContainerProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
	maxWidth: MaxWidth
}

const parseMaxWidth = (maxWidth: MaxWidth) => {
	switch(maxWidth) {
		case 'sm':
			return 'max-w-5xl'
		case 'md':
			return 'max-w-7xl'
		case 'lg':
			return 'max-w-container'
	}
}


const Container = React.forwardRef<HTMLDivElement, ContainerProps>(
	({ className, maxWidth = 'lg', ...rest }, ref) => {
		const styles = `container ${parseMaxWidth(maxWidth)} mx-auto px-4`;
		return (
			<div
				ref={ref}
				{...rest}
				className={RapidStyles(className, styles)}
			/>
		);
	},
);

export default Container;
