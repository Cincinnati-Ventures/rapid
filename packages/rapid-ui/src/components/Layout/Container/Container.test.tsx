import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import Container from './Container';

describe('Container', () => {
	it('should render default correctly', () => {
		const { container } = render(<Container />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(<Container styles='p-4' maxWidth='sm' />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with non default props', () => {
		const { container } = render(<Container styles='p-4' maxWidth='md' />);
		expect(container).toMatchSnapshot();
	});
});
