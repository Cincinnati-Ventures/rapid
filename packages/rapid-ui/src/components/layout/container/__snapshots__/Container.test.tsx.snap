// Vitest Snapshot v1

exports[`Container > should render correctly with non default props 1`] = `
<div>
  <div
    class="container max-w-5xl mx-auto p-4 rapid-container"
  />
</div>
`;

exports[`Container > should render correctly with non default props 2`] = `
<div>
  <div
    class="container max-w-7xl mx-auto p-4 rapid-container"
  />
</div>
`;

exports[`Container > should render default correctly 1`] = `
<div>
  <div
    class="container max-w-container mx-auto px-4 rapid-container"
  />
</div>
`;
